### PR TITLE
Add dataResidencyRegion option for sending to portal

### PR DESCRIPTION
**Pull request by MikhailSuendukov**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Add new API that allow to specify the data residency region for the current device.

## Related PRs or issues
[AB#102463](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/102463)

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/Device.java [hunk 1]
```diff
@@ -119,6 +123,11 @@ public class Device extends WrapperSdk {
      */
     private String carrierCountry;
 
+    /**
+     * Carrier data residency region.
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/Device.java:127]:
```suggestion
     * Data residency region.
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java [hunk 2]
```diff
@@ -308,6 +309,15 @@ public static void setCountryCode(String countryCode) {
         DeviceInfoHelper.setCountryCode(countryCode);
     }
 
+    /**
+     * Set the country code or any other string to identify residency region.
+     *
+     * @param dataResidencyRegion residency region code.
+     */
+    public static void setDataResidencyRegion(@Nullable String dataResidencyRegion) {
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java:317]: Why @Nullable here? Do we have any use case of `AppCenter.setDataResidencyRegion(null)`

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/DeviceInfoHelper.java [hunk 3]
```diff
@@ -211,6 +221,15 @@ public static void setCountryCode(String countryCode) {
         AppCenterLog.debug(AppCenterLog.LOG_TAG, String.format("Set country code: %s", countryCode));
     }
 
+    /**
+     * Set the country code or any other string to identify residency region.
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/DeviceInfoHelper.java:225]:
```suggestion
     * Set the country code or any other string to identify the data residency region.
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/DeviceInfoHelper.java [hunk 4]
```diff
@@ -211,6 +221,15 @@ public static void setCountryCode(String countryCode) {
...
+     *
+     * @param dataResidencyRegion residency region code.
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/DeviceInfoHelper.java:227]:
```suggestion
     * @param dataResidencyRegion data residency region code.
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/AbstractLog.java [hunk 5]
```diff
@@ -84,6 +92,11 @@ public abstract class AbstractLog implements Log {
      */
     private Device device;
 
+    /**
+     * Carrier data residency region.
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/AbstractLog.java:96]:
```suggestion
     * Data residency region.
```

**Review by DmitriyKirakosyan — APPROVED** [review]

**Comment by EatonZ** [comment]
@DmitriyKirakosyan @MikhailSuendukov Hi, is there any documentation for this new setting? In particular, what is the list of supported regions that can be set?

**Pull request merged**